hail_search/queries/base.py

```diff
@@ -679,18 +693,31 @@ def _filter_compound_hets(self):
         # Get possible pairs of variants within the same gene
         ch_ht = ch_ht.annotate(gene_ids=self._gene_ids_expr(ch_ht, comp_het=True))
         ch_ht = ch_ht.explode(ch_ht.gene_ids)
-        self.unfiltered_comp_het_ht = ch_ht
-
-        formatted_rows_expr = hl.agg.collect(ch_ht.row)
-        if self._has_secondary_annotations:
-            if HAS_ALLOWED_ANNOTATION not in self._ht.row or HAS_ALLOWED_SECONDARY_ANNOTATION not in self._ht.row:
-                # In cases where comp het pairs must have different data types, there are no single data type results
-                return None
-            primary_variants = hl.agg.filter(ch_ht[HAS_ALLOWED_ANNOTATION], formatted_rows_expr)
-            secondary_variants = hl.agg.filter(ch_ht[HAS_ALLOWED_SECONDARY_ANNOTATION], formatted_rows_expr)
+
+        # Filter allowed transcripts to the grouped gene
+        transcript_annotations = {
+            k: ch_ht[k].filter(lambda t: t.gene_id == ch_ht.gene_ids)
+            for k in [ALLOWED_TRANSCRIPTS, ALLOWED_SECONDARY_TRANSCRIPTS] if k in ch_ht.row
+        }
+        if transcript_annotations:
+            ch_ht = ch_ht.annotate(**transcript_annotations)
+        primary_filters = self._get_annotation_filters(ch_ht)
+        secondary_filters = self._get_annotation_filters(ch_ht, is_secondary=True)
+
+        self.unfiltered_comp_het_ht = ch_ht.filter(hl.any(primary_filters + secondary_filters))
+        if self._has_secondary_annotations and not (primary_filters and secondary_filters):
```

Comment:
This line is a little unclear to me.  I think right now `not (primary_filters and secondary_filters)` is `True` if either `primary_filters` or `secondary_filters` has length > 0.  Is that what we want?

Reply:
In english, this is saying that if there are supposed to be secondary filters, but we do not have both primary and secondary filters, then there are no valid pars for this data type and we should break. So `(primary_filters and secondary_filters)` is only true if both have length > 0 , so the `not` means that if either have length 0 we meet the condition